flexget/plugins/input/dynamic_imdb.py

```diff
@@ -112,130 +138,126 @@ def get_entity_type_and_object(self, imdb_id):
-                log.debug('Searching for movies that correlates to: ' + job_type)
-                items += person.get(job_type)
-                continue
+        if entity_type == 'Company':
+            return self.items_by_company(entity_object)
+
+        if entity_type == 'Character':
+            return self.items_by_character(entity_object, content_types)
+
+        elif entity_type == 'Person':
+            return self.items_by_person(entity_object, job_types, content_types)
+
+    def flatten_list(self, _list):
+        """
+        Gets a list of lists and returns a flat list
+        """
+        for el in _list:
+            if isinstance(el, collections.Iterable) and not isinstance(el, basestring):
+                for sub in self.flatten_list(el):
+                    yield sub
             else:
-                for content_type in content_types:
-                    job_and_content = job_type + ' ' + CONTENT_TYPE_CONVERSION[content_type]
-                    log.debug('Searching for movies that correlates to: ' + job_and_content)
-                    movies_by_job_type = person.get(job_and_content)
-                    if content_type == 'movie':
-                        log.debug('Searching for short movies and documentaries.')
-                        movies_by_job_type += person.get(job_type + ' ' + 'documentary', [])
-                        movies_by_job_type += person.get(job_type + ' ' + 'short', [])
-                    if movies_by_job_type:
-                        for item in movies_by_job_type:
-                            self.ia.update(item)
-                            if item not in items and item['kind'] in content_types:
-                                log.debug('Found item: ' + item.get('title') + ', adding to unfiltered list')
-                                items.append(item)
-                            else:
-                                log.debug('Item ' + str(item) + ' already found in list, skipping.')
-        return items
+                yield el
 
-    def get_items_by_character(self, character, content_types):
+    def no_empty_objects(self, _list):
         """
-        Return items list for a character object
-        :param character: character object
-        :param content_types: content types as defined in config
-        :return:
+        Returns list without None objects
         """
+        return [item for item in _list if item is not None]
+
+    def flat_list(self, non_flat_list, remove_none=False):
+        flat_list = self.flatten_list(non_flat_list)
+        if remove_none:
+            flat_list = self.no_empty_objects(flat_list)
+        return flat_list
+
+    def filtered_items(self, unfiltered_items, content_types):
         items = []
-        for content_type in content_types:
-            items += character.get(CHARACTER_CONTENT_TYPE_CONVERSION[content_type], [])
+        unfiltered_items = list(set(unfiltered_items))
```

Comment:
No need to turn this back into a list here, `sorted` can operate on any iterable.